WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartAdapter.kt

```diff
@@ -2,138 +2,43 @@ package org.wordpress.android.ui.quickstart
 
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
-import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView.ViewHolder
-import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTask
-import org.wordpress.android.ui.quickstart.viewholders.CompletedHeaderViewHolder
+import org.wordpress.android.ui.quickstart.QuickStartFullScreenDialogFragment.QuickStartTaskCard
 import org.wordpress.android.ui.quickstart.viewholders.TaskViewHolder
 
-class QuickStartAdapter(
-    tasksUncompleted: MutableList<QuickStartTask?>,
-    tasksCompleted: MutableList<QuickStartTask?>,
-    isCompletedTasksListExpanded: Boolean
-) : Adapter<ViewHolder>() {
-    private val tasks: MutableList<QuickStartTask?> = mutableListOf()
-    private val tasksUncompleted: MutableList<QuickStartTask?>
-    private val taskCompleted: MutableList<QuickStartTask?>
-    private var listener: OnQuickStartAdapterActionListener? = null
-    var isCompletedTasksListExpanded: Boolean
-        private set
-
-    init {
-        tasks.addAll(tasksUncompleted)
-        if (tasksCompleted.isNotEmpty()) {
-            tasks.add(null) // adding null where the complete tasks header simplifies a lot of logic for us
-        }
-        this.isCompletedTasksListExpanded = isCompletedTasksListExpanded
-        if (this.isCompletedTasksListExpanded) {
-            tasks.addAll(tasksCompleted)
-        }
-        this.tasksUncompleted = tasksUncompleted
-        this.taskCompleted = tasksCompleted
-    }
-
+class QuickStartAdapter : ListAdapter<QuickStartTaskCard, ViewHolder>(QuickStartAdapterDiffCallback) {
     override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int) = when (viewType) {
```

Comment:
❓ @ashiagr Since there is only one kind of view holder present in the adapter now, does it make sense for us to replace this `ViewHolder` with `TaskViewHolder`?.

Reply:
Good catch! Actually, I decided to not make this change and also keep the when clause in the adapter in spite of having just one view holder as we might want to introduce another view holder for the completed task with a checkmark according to the new visuals: pbArwn-4wS-p2; then we'll have to bring back this logic. In case I can reuse the existing view holder without having to create another one, I'll refactor this part of the logic in an upcoming PR.

Reply:
Thanks for making it clear. 👍🏼 . Makes sense. I thought that would be the case, still wanted to confirm it. 🙌🏼 .